src/atom-environment.coffee

```diff
@@ -678,6 +684,7 @@ class AtomEnvironment extends Model
         @deserialize(state) if state?
         @deserializeTimings.atom = Date.now() - startTime
 
+        @document.body.appendChild(@views.getView(@titleBar)) if @titleBar
```

Comment:
Would it make sense to add the title bar into the `atom-panel-container.header` with [addHeaderPanel()](https://atom.io/docs/api/v1.7.3/Workspace#instance-addHeaderPanel).

Then you _could_ do some funky stuff like (visually) combining the title-bar with the tool-bar:

![screen shot 2016-05-20 at 5 30 32 pm](https://cloud.githubusercontent.com/assets/378023/15445150/2a61fa56-1f35-11e6-98ba-25c6101d2f42.png)

Probably not by default, but maybe just restyle it in `styles.less` or as an option in the tool-bar package.